examples/lib/stories/experimental/loading_screen.dart

```diff
@@ -0,0 +1,239 @@
+      progressNotifier.reportLoadingProgress(
+        ProgressMessage(
+          message: 'The notification message #$messagesSent',
+          progress: (anyProgressEmulation * 100).toInt(),
+        ),
+      );
+    }
+    super.update(dt);
+  }
+}
+
+/// A message strict type. You should to define an immutable message class for
+/// exchanging between game and components. You also could to use just [String]
+/// and even [dynamic], but the last option looks like heavy anti-pattern.
+@immutable
+class ProgressMessage {
+  const ProgressMessage({required this.message, required this.progress});
+
+  /// Just for fast illustration - a special constructor to indicate, that
+  /// [Game.onLoad] is finished.
+  const ProgressMessage.finished()
+      : message = '',
+        progress = 101;
+
+  final String message;
+  final int progress;
+}
+
+/// The loading screen widget with animated progress bar anf fade effect
+/// when displaying the game.
+class LoadingScreenExampleWidget extends StatefulWidget
+    with LoadingWidgetMixin<ProgressMessage> {
+  LoadingScreenExampleWidget({super.key});
+
+  @override
+  State<LoadingScreenExampleWidget> createState() =>
+      _LoadingScreenExampleWidgetState();
+}
+
+class _LoadingScreenExampleWidgetState extends State<LoadingScreenExampleWidget>
+    with SingleTickerProviderStateMixin {
+  late AnimationController _controller;
+  String text = '';
+  double value = 0;
+  bool showGame = false;
+
+  /// Message listener should be initialized here. The best place is the end of
+  /// function to minimize a chance to loose an message, because everything is
+  /// asynchronous
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(vsync: this);
+    _controller.addListener(() => setState(() {}));
+
+    /// Just call this and everything will work!
+    widget.initMessageListener(onProgressMessage);
+  }
+
+  /// Process received messages here.
+  /// This is a place for calling [setState] or whatever you prefer to
+  /// use for widget's state management.
+  void onProgressMessage(ProgressMessage message) {
+    setState(() {
+      if (message.progress == 101) {
+        showGame = true;
+        _controller.value = 0;
+        _controller.animateTo(
+          1,
+          duration: const Duration(seconds: 3),
+          curve: Curves.easeOut,
+        );
+      } else {
+        value = message.progress / 100;
+        text = message.message;
+        _controller.animateTo(
+          value,
+          duration: const Duration(seconds: 1),
+          curve: Curves.easeIn,
+        );
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    /// This is the game's widget. You should manually decide, when it is the
+    /// time to show it. But in return you have ability to add some effects!
+    if (showGame) {
+      return Opacity(
```

Comment:
Would wrapping the `FlameGame` in `Opacity` decrease performance even when the value is 1.0?
I remember `Opacity` being very cpu intensive: https://docs.flutter.dev/perf/best-practices#minimize-use-of-opacity-and-clipping
Maybe using a `AnimatedCrossFade` is more performant?

Reply:
Maybe, it was just fast insight when I added it - to demonstrate possibilities. 
You even could rebuild the widget without Opacity or any unnecessary widgets at all, after the animation finish

Reply:
I changed it to `AnimatedSwitcher` now.